gunpowder/roi.py

```diff
@@ -223,7 +223,21 @@ def intersect(self, other):
         '''Get the intersection of this ROI with another :class:`Roi`.'''
 
         if not self.intersects(other):
-            return Roi(shape=(0,)*self.dims()) # empty ROI
+            # preserve the offset if the two non-intersecting ROIs share it
```

Comment:
Thanks Ben. We should handle 0 volume cases better. Do we need to have logic handling empty cases specially though? I think I would expect `self.intersects(other)` to return true if `self` is `Roi((0,0,0), (10,10,10))` and other is `Roi((5,5,5),(0,0,0))`. Would removing lines 197, 198 be sufficient? 
``` python
if self.empty() or other.empty():
    return False
```

Reply:
@funkey looks like these lines were added in 1297d7c39f21fdaad4bf271327865392ca67e76b to fix handling of empty roi's. How this would break without them is unclear to me though.

Reply:
As discussed with @pattonw offline, this part is fine: Empty ROIs should not intersect with anything (since the intersection would be empty, and that is our criterion for determining whether two ROIs intersect).